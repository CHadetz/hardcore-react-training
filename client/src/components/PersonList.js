import React from "react";
import Person from "./Person";

const PersonList = props => {
  const { persons, firePerson } = props;

  if (persons.count() === 0) {
    return false;
  }

  const averageAge = persons.reduce((r, p) => r + p.age, 0) / persons.count();

  return (
    <div>
      <h3>
        Average age: {averageAge.toFixed(3)}
      </h3>

      {persons.map(p =>
        <Person firePerson={firePerson} person={p} key={p.id} />
      )}
    </div>
  );
};

export default PersonList;
