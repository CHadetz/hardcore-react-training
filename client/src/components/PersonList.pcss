.list {
  columns: 15em;
}
