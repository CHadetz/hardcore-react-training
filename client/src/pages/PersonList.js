import React from "react";
import Person from "./Person";

const PersonList = props => {
  const { persons, onDelete } = props;

  const sorted = persons.sortBy(p => p.firstName).sortBy(p => p.lastName);
  /*
    .filter(p => p.age < 30)
    .filter(p => p.gender === "m");
    */

  return (
    <div>
      {sorted.map(p => <Person key={p.id} person={p} onDelete={onDelete} />)}
    </div>
  );
};

export default PersonList;
