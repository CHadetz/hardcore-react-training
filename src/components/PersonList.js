// @flow

import React from 'react';
import styles from './PersonList.pcss';
import Person from './Person';
import ImmutablePropTypes from 'react-immutable-proptypes';

const PersonList = props => {

  const { persons, title, ...rest } = props;

  if (persons.length === 0) {
    return false;
  }

  const averageAge = persons.reduce((r, p) => r + p.age, 0) / persons.count();
  return (
    <div className={styles.root}>

      <h2>{title}</h2>

      <p>
        Keski-ik√§: {averageAge.toFixed(2)}
      </p>

      {persons.map(person => <Person key={person.id} person={person} {...rest} />)}
    </div>
  );
};

PersonList.propTypes = {
  persons: ImmutablePropTypes.list.isRequired,
  title: React.PropTypes.string.isRequired,
};


export default PersonList;
