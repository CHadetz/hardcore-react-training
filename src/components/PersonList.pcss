.root {
  border: 5px dotted #FF00FF;
  padding: 1em;
}
