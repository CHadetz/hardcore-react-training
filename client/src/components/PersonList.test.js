import React from 'react';
import PersonList from './PersonList';
import renderer from 'react-test-renderer';
import { List } from 'immutable';

const persons = List.of({
  id: 'feikki-id',
  firstName: 'Pekka',
  lastName: 'Naator',
  age: 40,
  gender: 'm',
});

const firePerson = () => {

};

it('renders correctly', () => {
  const tree = renderer.create(
    <PersonList persons={persons} firePerson={firePerson} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
