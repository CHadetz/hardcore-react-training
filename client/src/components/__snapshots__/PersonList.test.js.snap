// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h3>
    Average age: 
    40.000
  </h3>
  <div
    className="person male"
  >
    <strong>
      Naator
    </strong>
    , 
    Pekka
    <div>
      m
      , 
      40
    </div>
    <button
      className="sc-bdVaJa hlmQtC"
      onClick={[Function]}
    >
      :) Erota :)
    </button>
  </div>
</div>
`;
